config/routes.rb

```diff
@@ -3,18 +3,35 @@
   # See how all your routes lay out with "rake routes".
 
   # You can have the root of your site routed with "root"
-  # root 'welcome#index'
+  # root 'task#index'
   get '/' => 'task#index'
+  # lines 6 and 7 do the exact same thing
+  get '/new' => 'task#new'
+  post '/tasks' => 'task#create'
 
-  get '/show/:id' => 'task#show'
-  post '/show/:id' => 'task#show'
+  get '/tasks/update/:id' => 'task#edit'
```

Comment:
For RESTful routing, you'd want to use `/tasks/:id/update` instead, putting the ID between the controller and the action